### PR TITLE
Remove edit link from links to bridgeheads

### LINKED ISSUE #903: Extra problematic edit link
There's an extra “edit” link at the very top of this file: https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-put-job.html

image

If you click it, it takes you to an inappropriate location.  If I click the next lower link, it takes me to the right place (https://raw.githubusercontent.com/elastic/elasticsearch/master/docs/reference/ml/apis/put-job.asciidoc).

The pages before and after this one don’t seem to have the problematic edit link in the title section.  There are no overrides specified for this file (it’s included here: https://raw.githubusercontent.com/elastic/elasticsearch/master/docs/reference/ml/apis/ml-api.asciidoc).

NOTE: The same problem occurs when you use the --asciidoctor build option too.

### PR DESCRIPTION
**Pull request by nik9000**
We had a bug where links to floating titles (aka bridgeheads) would
include the `edit_me` link in its contents like this:

```
<a class="xref" href="chapter.html#floater" title="Floateredit">
  Floater
  <a href="https://github.com/elastic/docs/edit/master/index.asciidoc" class="edit_me" title="Edit this page on GitHub" rel="nofollow">
    edit
  </a>
</a>
```

Because this is invalid html it'd cause the `edit` button to float to
the top of the page, and, confusingly, take you to edit the source of
the *link target* rather than the current page. Madness.

This removes the extra link.

Closes #903

### CODE REVIEW

**Comment by nik9000** [comment]
@lcawl, could you check this one when you have a chance? I don't think we really have a code reviewer for the xslt bit of the change. It is sort of a evolutionary dead end. Especially for working with such large bits of xml.

We'll have to manually rebuild all the docs after merging this to fix all of the funny edit links.

File: integtest/spec/single_book_spec.rb [hunk 1]
```diff
@@ -145,6 +145,25 @@
         end
       end
     end
+    context 'for a book that a reference to a floating title' do
```
Comment (lcawl) [inline integtest/spec/single_book_spec.rb]:
```suggestion
    context 'for a book that has a reference to a floating title' do
```

**Comment by lcawl** [comment]
@alaudazzi Found what seem to be similar problems in https://www.elastic.co/guide/en/ecs/current/ecs-guidelines.html and https://www.elastic.co/guide/en/elastic-stack-overview/current/how-watcher-works.html

**Comment by nik9000** [comment]
I've merged this. I'll kick off a full rebuild of the docs later on. I can't do it right now because we're upgrading jenkins and it is in "shutting down" mode and won't take new jobs.

**Pull request merged**